File: go/day_6/lanternfish.go
package main

func SchoolSize(initial_state []uint64, days int) uint64 {
	var school [9]uint64
	var count uint64

	for i := 0; i < len(initial_state); i++ {
		school[initial_state[i]] += 1
	}

	for i := 0; i < days; i++ {
		zeros := school[0]
		for j := 1; j < len(school); j++ {
			school[j-1] = school[j]
		}
		school[8] = zeros
		school[6] += zeros
	}

	for j := 1; j < len(school); j++ {
		count += school[j]
	}

	return count
}

File: typescript/src/day_6/lanternfish.ts
export const schoolSize = (initial_state: number[], days: number) : number => {
  const school: number[] = new Array(9).fill(0);
  initial_state.map(state => {
    school[state] = school[state] ? school[state] + 1 : 1;
  })

  for (const _day of Array(days).keys()) {
    let zeros = school.shift();
    if(zeros !== undefined) {
      school.push(zeros);
      school[6] += zeros;
    }
  }

  return school.reduce((acc, value) => acc + value);
}
